node/src/components/block_proposer.rs

```diff
@@ -410,47 +382,26 @@ impl BlockProposerReady {
         info!(%hash, "added deploy to the buffer");
     }
 
-    /// Notifies the block proposer that a block has been finalized.
-    fn finalized_deploys<I>(&mut self, deploys: I)
-    where
-        I: IntoIterator<Item = DeployOrTransferHash>,
-    {
-        for deploy_hash in deploys.into_iter() {
-            let (hash, remove_result) = match deploy_hash {
-                DeployOrTransferHash::Deploy(hash) => {
-                    (hash, self.sets.pending_deploys.remove(&hash))
-                }
-                DeployOrTransferHash::Transfer(hash) => {
-                    (hash, self.sets.pending_transfers.remove(&hash))
-                }
+    /// Handles finalization of a block.
+    fn handle_finalized_block(&mut self, block: &FinalizedBlock) -> Effects<Event> {
+        for deploy_hash in block.deploy_hashes() {
+            let expiry = match self.sets.pending_deploys.remove(deploy_hash) {
```

Comment:
Why do we need to keep the collections for `pending_deploys` and `pending_transfers` separate?

Reply:
The deploy and transfer hashes are separate entries in the block, so it's convenient for proposing blocks if they're already separate. (We could also merge them, and only separate them when proposing… either way, it's not something that was introduced in this PR.)